src/DotNetWorker.Core/Hosting/CoreWorkerHostBuilderExtensions.cs

```diff
@@ -38,5 +40,35 @@ public static IHostBuilder ConfigureFunctionsWorker(this IHostBuilder builder, A
 
             return builder;
         }
+
+        /// <summary>
+        /// Invokes auto-generated configuration methods for a given <see cref="IHostBuilder"/>.
+        /// This method searches for classes that implement the <see cref="IAutoConfigureStartup"/> interface,
+        /// excluding interfaces and abstract classes. For each identified class, it locates the
+        /// <c>Configure</c> method with the signature <c>void Configure(IHostBuilder hostBuilder)</c>,
+        /// and executes the method using an instance of the class.
+        /// </summary>
+        /// <param name="builder">The <see cref="IHostBuilder"/> to configure.</param>
+        /// <returns>The same <see cref="IHostBuilder"/> after the auto-generated configuration methods are invoked.</returns>
+        public static IHostBuilder InvokeAutoGeneratedConfigureMethods(this IHostBuilder builder)
+        {
+            var autoConfigureStartupTypes = Assembly.GetEntryAssembly()!
+                .GetTypes()
+                .Where(t => typeof(IAutoConfigureStartup).IsAssignableFrom(t) && !t.IsInterface && !t.IsAbstract);
+
+            foreach (var type in autoConfigureStartupTypes)
+            {
+                var configureMethod = type.GetMethod(nameof(IAutoConfigureStartup.Configure), new[] { typeof(IHostBuilder) });
+                if (configureMethod == null)
+                {
+                    continue;
+                }
+
+                var instance = Activator.CreateInstance(type) as IAutoConfigureStartup;
```

Comment:
small preference for a hard cast - if this is somehow `null`, something has gone wrong.

Reply:
That makes sense. Switched to direct cast.